emotion/app/root.tsx

```diff
@@ -35,21 +35,28 @@ const Document = withEmotionCache(
   ({ children, title }: DocumentProps, emotionCache) => {
     const serverStyleData = useContext(ServerStyleContext);
     const clientStyleData = useContext(ClientStyleContext);
+    const reinjectStylesRef = useRef(true);
 
     // Only executed on client
+    // When a top level ErrorBoundary or CatchBoundary are rendered,
+    // the document head gets removed, so we have to create the style tags
     useEffect(() => {
-      // re-link sheet container
-      emotionCache.sheet.container = document.head;
+      if (reinjectStylesRef.current) {
```

Comment:
What do you think about using an early return here?

```suggestion
      if (!reinjectStylesRef.current) {
        return;
      }
```